hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -656,7 +659,9 @@ private List<DirectoryInfo> listAllPartitions(HoodieTableMetaClient datasetMetaC
           partitionsToBootstrap.add(dirInfo);
         } else {
           // Add sub-dirs to the queue
-          pathsToList.addAll(dirInfo.getSubDirectories());
+          pathsToList.addAll(dirInfo.getSubDirectories().stream()
+              .map(path -> new SerializablePath(new CachingPath(path.toUri())))
+              .collect(Collectors.toList()));
         }
```

Comment:
unrelated changes?

Reply:
yeah my bad. mixed up w/ another patch. 
will revert it.